### PR TITLE
Deus Ex Revival Attempt

### PR DESCRIPTION
**Pull request by gbasood**
Original PR here: #5669 
In which I do a merge commit and probably fuck shit up

@DrCelt This changes things related to hypothermia handling so I'll need you to review it to tell me how to re-implement that shit, especially where it comes to plasmen

### CODE REVIEW

**Comment by Imagebard** [comment]
I LIVE
*I DIE*
**I LIVE AGAIN**

:+1: :+1: :+1: :+1: :+1:

**Comment by CptWad** [comment]
https://www.youtube.com/watch?v=ZCk8ByyCsks

File: code/modules/organs/organ_objects.dm [hunk 1]
```diff
@@ -101,13 +101,29 @@
...
 	organ_tag = "lungs"
 	organ_type = /datum/organ/internal/lungs
 
+/obj/item/organ/lungs/vox
+	name = "vox lungs"
+	icon_state = "vox-lungs"
+	organ_type = /datum/organ/internal/lungs/vox
+
+/obj/item/organ/lungs/plasmaman
+	name = "weird pink lungs"
```
Comment (9600bauds) [inline code/modules/organs/organ_objects.dm:117]: reeee

File: html/changelogs/N3X15-deus-ex.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+author: N3X15
+delete-after: true
+changes:
+  - experiment: "Deus Ex Update:"
+  - rscadd: Lungs are now fully modular and actually process gas for humans.  Lungs must be compatible with a species or they will die of suffocation.
+  - rscadd: Vox and Plasmamen have their own lungs.  These alien lungs are not available at organ printers, requiring stockpiling from donor monkey-humans.
```
Comment (9600bauds) [inline html/changelogs/N3X15-deus-ex.yml]: Looks like they are available

**Comment by 9600bauds** [comment]
I kind of skimmed over this and this code looks pretty damn clean honestly. :+1: from me.

**Comment by gbasood** [comment]
N3X did gud
I'll work on those comments.

**Comment by ghost** [comment]
I had an idea for an organ harvester for medbay to replace the organic organ synthesiser. Never got around to coding it but could totally have a look at it again for the PR, could be pretty neat.

File: html/changelogs/N3X15-deus-ex.yml (hunk 2 above)
Comment (ihadtoregisterforthis) [inline html/changelogs/N3X15-deus-ex.yml]: Even if they weren't, there wouldn't be a way of getting plasmamen lungs like this.

**Comment by gbasood** [comment]
I renamed plasmen lungs but github didn't notice I guess

**Comment by gbasood** [comment]
Alright, so this makes the deus ex PR, as it was before, work as desired.

But mechanical organs at roundstart still do fucking nothing
input needed

**Comment by gbasood** [comment]
Actually, fuck it, mechanical organs can happen in another PR.

**Comment by Exxion** [comment]
When

File: code/modules/mob/living/carbon/species.dm [hunk 3]
```diff
@@ -137,6 +137,19 @@ var/global/list/whitelisted_species = list("Human")
 					return 0
 	return 1
 
+/datum/species/proc/clear_organs(var/mob/living/carbon/human/H)
+	if(H.organs)
+		H.organs.len=0
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/species.dm:142]: Was that memory leak ever fixed?

File: code/modules/mob/living/carbon/species.dm [hunk 4]
```diff
@@ -175,154 +191,25 @@ var/global/list/whitelisted_species = list("Human")
...
-		else
-			H.adjustOxyLoss(HUMAN_MAX_OXYLOSS)
-			H.failed_last_breath = 1
-		H.oxygen_alert = max(H.oxygen_alert, 1)
-
-	else								// We're in safe limits
+// Sent from /datum/lung_gas/metabolizable.
+/datum/species/proc/receiveGas(var/gas_id, var/ratio, var/moles, var/mob/living/carbon/human/H)
+	//testing("receiveGas: [gas_id] ? [breath_type] - ratio=[ratio], moles=[moles]")
+	if(ratio <= 0 || gas_id != breath_type)
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/species.dm:197]: ~~... And what if there's multiple gasses in the air? Which does it pick?~~

File: code/modules/mob/living/carbon/species.dm [hunk 5]
```diff
@@ -574,6 +493,15 @@ var/global/list/whitelisted_species = list("Human")
 //	back_icons      = 'icons/mob/back.dmi'
 
 	has_mutant_race = 0
+	has_organ = list(
+		"heart" =    /datum/organ/internal/heart,
+		"lungs" =    /datum/organ/internal/lungs/vox, // VOX LUNGS
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/species.dm]: Totally couldn't figure that out Mr commenter.

File: code/modules/organs/internal/liver/liver.dm [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+			if (src.damage < min_broken_damage)
+				src.damage += 0.2 * process_accuracy
+			//Damaged one shares the fun
+			else
+				var/datum/organ/internal/O = pick(owner.internal_organs)
+				if(O)
+					O.damage += 0.2  * process_accuracy
+
+		//Detox can heal small amounts of damage
+		if (src.damage && src.damage < src.min_bruised_damage && owner.reagents.has_reagent("anti_toxin"))
```
Comment (PJB3005) [inline code/modules/organs/internal/liver/liver.dm:43]: The anti toxin check is an OOP crime

File: code/modules/organs/internal/liver/liver.dm [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+			src.damage -= 0.2 * process_accuracy
+
+		// Damaged liver means some chemicals are very dangerous
+		if(src.damage >= src.min_bruised_damage)
```
Comment (PJB3005) [inline code/modules/organs/internal/liver/liver.dm:47]: Holy shit son snowflake shit non OOP code much?

File: code/modules/organs/internal/liver/liver.dm (hunk 6 above)
Comment (ghost) [inline code/modules/organs/internal/liver/liver.dm:43]: also you could use owner.has_reagent_in_blood instead

File: code/modules/organs/internal/lungs/lung.dm [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+///////////////////////
+// LUNG ORGAN
+///////////////////////
+
+/datum/organ/internal/lungs
+	name = "lungs"
+	parent_organ = "chest"
+	removed_type = /obj/item/organ/lungs
+
+	// VG now delegates breathing to the appropriate organ.
```
Comment (ghost) [inline code/modules/organs/internal/lungs/lung.dm]: > VG

rrrrrrrrrrrrrrreeeeeeeeeeeeeeeee

File: code/modules/organs/internal/liver/liver.dm (hunk 6 above)
Comment (9600bauds) [inline code/modules/organs/internal/liver/liver.dm:43]: To be fair, this code predates that proc

File: code/modules/organs/internal/liver/liver.dm (hunk 7 above)
Comment (9600bauds) [inline code/modules/organs/internal/liver/liver.dm:47]: What's wrong tho

File: code/modules/organs/internal/liver/liver.dm (hunk 6 above)
Comment (ghost) [inline code/modules/organs/internal/liver/liver.dm:43]: Get with the times, grandpa

File: code/modules/mob/living/carbon/species.dm (hunk 3 above)
Comment (gbasood) [inline code/modules/mob/living/carbon/species.dm:142]: I can't reproduce a memory leak of any kind with either len = 0 or Cut() so yes

File: code/modules/mob/living/carbon/species.dm (hunk 4 above)
Comment (gbasood) [inline code/modules/mob/living/carbon/species.dm:197]: That's not how that works

File: code/modules/organs/internal/liver/liver.dm (hunk 6 above)
Comment (gbasood) [inline code/modules/organs/internal/liver/liver.dm:43]: How SHOULD it be done now, then

File: code/modules/organs/internal/lungs/lung.dm (hunk 8 above)
Comment (gbasood) [inline code/modules/organs/internal/lungs/lung.dm]: USEFUL INPUT 10/10 THANKS CELT

Comment (ghost) [inline code/modules/organs/internal/lungs/lung.dm]: it's /vg/ though

File: code/modules/organs/internal/liver/liver.dm (hunk 6 above)
Comment (PJB3005) [inline code/modules/organs/internal/liver/liver.dm:43]: @gbasood in the reagent's code.

File: code/modules/organs/internal/lungs/lung.dm [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+	)
+
+	var/inhale_volume = BREATH_VOLUME
+	var/exhale_moles = 0
+
+/datum/organ/internal/lungs/proc/gasp()
+	owner.emote("gasp")
+
+/datum/organ/internal/lungs/proc/handle_breath(var/datum/gas_mixture/breath, var/mob/living/carbon/human/H)
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/datum/species/proc/handle_breath() called tick#: [world.time]")
```
Comment (PJB3005) [inline code/modules/organs/internal/lungs/lung.dm]: Ah I spent a couple hours writing a script to remove these, remember?

File: code/modules/organs/internal/lungs/lung.dm (hunk 8 above)
Comment (Imagebard) [inline code/modules/organs/internal/lungs/lung.dm]: So when are you going to follow celt's advice and replace VG with /vg/?

File: html/changelogs/N3X15-deus-ex.yml [hunk 10]
```diff
@@ -0,0 +1,13 @@
+author: N3X15
```
Comment (PJB3005) [inline html/changelogs/N3X15-deus-ex.yml]: Hey hey gotta get them codervalids.

Comment (ghost) [inline html/changelogs/N3X15-deus-ex.yml]: Nah, let people think he's back

Comment (gbasood) [inline html/changelogs/N3X15-deus-ex.yml]: The literal next line says 'I just stole this'
Because I did

**Comment by Exxion** [comment]
HAP
PEN
ING

**Comment by ghost** [comment]
Oh lord jesus

**Comment by Imagebard** [comment]
YOUR TIME IS UP, MY TIME IS NOW.

**Comment by ghost** [comment]
100% TESTED

**Comment by TheEliminator16** [comment]
Sweet maybe now we can do a full cybernetic update with cool new cyber limbs and st-
Ok

**Comment by ghost** [comment]
I'd like to debunk this a little with the fact you can still die from heat

Might only be affecting those who join on roundstart?

**Comment by TheEliminator16** [comment]
What fucked up?

**Comment by ghost** [comment]
People can't bleed out, oxygen damage doesn't count for squat now for some reason

**Comment by ghost** [comment]
I received no damage in plasmafire without suit, got stunned once and never stood up for rest of round, got spaced and still got no damage.

**Comment by TheEliminator16** [comment]
We don't want people dying on this server, we're full on furry ERP :^)

**Comment by gbasood** [comment]
Something in life code completely fucked up
And I don't know why because there were no runtimes

**Comment by ghost** [comment]
:100:%

**Comment by ihadtoregisterforthis** [comment]
Here lies Nexis' legacy

**Comment by SysadminJeroen** [comment]
:100:% tested yep TOTALLY WORKS ALRIGHT!

**Comment by gbasood** [comment]
So the problem was that N3X15 didn't update the proc call to liver's metabolize reagent proc
So it was trying to multiply "ryetalin" by 1 instead of another number by 1 and runtiming life code

**Comment by Exxion** [comment]
Didn't you say there weren't any runtimes?

**Comment by gbasood** [comment]
@Exxion That's what PJB told me. I checked the logs myself post-round and there were.

**Comment by PJB3005** [comment]
I was looking for ones specific to the welding helmets...

**Comment by gbasood** [comment]
:+1:

**Pull request merged**